java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/debugging/NbProtocolServer.java

```diff
@@ -286,11 +290,33 @@ public NbProtocolServer(DebugAdapterContext context) {
                     stackFrame.setId(frameId);
                     stackFrame.setName(frame.getName());
                     URI sourceURI = frame.getSourceURI();
-                    if (sourceURI != null && sourceURI.getPath() != null) {
+                    if (sourceURI != null) {
                         Source source = new Source();
-                        source.setName(Paths.get(sourceURI).getFileName().toString());
-                        source.setPath(sourceURI.getPath());
-                        source.setSourceReference(0);
+                        String scheme = sourceURI.getScheme();
+                        if (null == scheme || scheme.isEmpty() || "file".equalsIgnoreCase(scheme)) {
+                            source.setName(Paths.get(sourceURI).getFileName().toString());
+                            source.setPath(sourceURI.getPath());
+                            source.setSourceReference(0);
+                        } else {
```

Comment:
A test for this path would be nice. Maybe extend `ServerTest`?

Reply:
Right. I've created https://issues.apache.org/jira/browse/NETBEANS-5084